plugins/brimcap/brimcap-plugin.ts

```diff
@@ -0,0 +1,240 @@
+        })
+    }
+
+    // add brimcap buttons to search and detail toolbars
+    this.api.toolbar.add("search", {
+      ...itemOptions,
+      id: searchButtonId,
+      command: brimcapDownloadSelectedCmd
+    })
+    this.api.toolbar.add("detail", {
+      ...itemOptions,
+      id: detailButtonId,
+      command: brimcapDownloadCurrentCmd
+    })
+
+    // add click handlers for button's emitted commands
+    this.cleanupFns.push(
+      this.api.commands.add(brimcapDownloadSelectedCmd, () => {
+        // this.api.toast("brimcapDownloadSelectedCmd!")
+
+        this.selectedConn && this.downloadPcap(this.selectedConn)
+      }),
+      this.api.commands.add(
+        brimcapDownloadCurrentCmd,
+        () => this.currentConn && this.downloadPcap(this.currentConn)
+      )
+    )
+
+    // add brim-command listeners to update button statuses
+    this.cleanupFns.push(
+      // the detail window's packets button will operate off of the 'current' record
+      this.api.commands.add("data-detail:current", ([record]) => {
+        if (!record) return
+        const data = Record.deserialize(record)
+
+        updateButtonStatus(
+          "detail",
+          detailButtonId,
+          data,
+          (conn) => (this.currentConn = conn)
+        )
+      }),
+      // the search window's packets button operates off of the 'selected' record
+      // (whatever is highlighted in the viewer/table)
+      this.api.commands.add("data-detail:selected", ([data]) => {
+        if (!data) return
+        updateButtonStatus(
+          "search",
+          searchButtonId,
+          data,
+          (conn) => (this.selectedConn = conn)
+        )
+      })
+    )
+  }
+
+  private logToSearchOpts(log: zng.Record): searchOptions {
+    const {spaceId} = this.api.getCurrent()
+
+    const ts = log.get("ts") as zng.Primitive
+    // RFC3999nano format with zero timezone offset
+    const formatter = DateTimeFormatter.ofPattern(
+      "yyyy-MM-dd'T'HH:mm:ss.SSSSSSSSS'Z'"
+    )
+    const tsString = LocalDateTime.ofEpochSecond(
+      getSec(ts),
+      getNs(ts),
+      ZoneOffset.UTC
+    )
+      .format(formatter)
+      .toString()
+
+    const dur = log.get("duration") as zng.Primitive
+    const dest = join(os.tmpdir(), `packets-${ts.toString()}.pcap`)
+    const root = path.join(this.dataRoot, spaceId)
+
+    return {
+      dstIp: log.get("id.resp_h").toString(),
+      dstPort: log.get("id.resp_p").toString(),
+      duration: `${dur.toFloat()}s`,
+      proto: log.get("proto").toString(),
+      root,
+      srcIp: log.get("id.orig_h").toString(),
+      srcPort: log.get("id.orig_p").toString(),
+      ts: tsString,
+      write: dest
+    }
+  }
+
+  private async downloadPcap(log: zng.Record) {
+    const searchOpts = this.logToSearchOpts(log)
+
+    const searchAndOpen = async () => {
+      await this.cli.search(searchOpts)
+      return open(searchOpts.write, {newWindow: true})
+    }
+
+    toast.promise(
```

Comment:
whoops, this should come from the api. I'll update this in the next commit with the ingest handling